ui/app/components/mount-backend-form.ts

```diff
@@ -24,9 +30,17 @@ import { isAddonEngine, allEngines } from 'vault/helpers/mountable-secret-engine
  *
  */
 
-export default class MountBackendForm extends Component {
-  @service store;
-  @service flashMessages;
+type MountModel = MountSecretBackendModel | AuthEnableModel;
```

Comment:
looks like this component is only rendered on 2 routes:

<img width="605" alt="Screenshot 2024-06-03 at 6 25 34 PM" src="https://github.com/hashicorp/vault/assets/903288/0f3cf1b6-3483-4fa8-ba4e-3ac03a252e6c"> 

in both instances, the component directly receives the result of the model hook as its argument for `@mountModel`. this is why i set up the types this way. 

initially, i thought about setting the type to be either `SecretEngineModel` or `AuthMethodModel`. however, this setup provides more flexibility -- it allows for potential changes in the route model that may include properties not specifically part of the underlying Ember data models.